plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/discoverer/XcpServerDiscoverer.java

```diff
@@ -529,13 +540,12 @@ public boolean processCommands(long agentId, long seq, Command[] commands) {
     }
 
     private void createXsToolsISO() {
-        String isoName = "xs-tools.iso";
-        VMTemplateVO tmplt = _tmpltDao.findByTemplateName(isoName);
+        VMTemplateVO tmplt = _tmpltDao.findByTemplateName(_toolsIsoName);
```

Comment:
The problem is at this line `_toolsIsoName` is not initialized yet (basically `cretaeXsToolsISO` is executed before newly added `determineToolsIsoName` method). My other idea is at this line we can register both `xs-tools.iso` and `guest-tools.iso` and either:
- later on (for example in `createServerResource` method) delete/deactivate the _wrong_ iso 
- present both of them on UI with different names and let the user select the _correct_ iso based on the version of their Xen

@rhtyd @wido @syed @rafaelweingartner @swill any comment, suggestion, concern?

Reply:
I would start with a different question. Do we need to check which XenServers versions are used? I mean, Can’t we simply persist both ISOs in the database? Without needing to use that “determine” method.

It seems that as long as we have both entries in the database (or new ones in the far far future ahead of us), your logic in ` CitrixResourceBase.java ` would take care of the rest. I would recommend one thing though. You have the strings "guest-tools" and "xs-tools" in two different classes (at least the ones I am seeing in this PR). What about extracting them to an enum?

Reply:
Yes we do, that's the whole point of this PR. The tools ISO has been renamed. So the question will be we _auto detect_ it or provide the options for user to select.

+1 on extracting to enum though, will do that.

Reply:
Can't we create a more transparent mode? I mean, the user sees just the "Xen-server-tools", and then underneath we select the right name according to the XenServer version.

Reply:
That's what I was trying to do but the way I did it was wrong, hence the first option mentioned above:
 > we can register both xs-tools.iso and guest-tools.iso and ... later on (for example in createServerResource method) delete/deactivate the wrong iso